torch_xla/torch_xla.py

```diff
@@ -56,9 +56,20 @@ def device_count() -> int:
   return len(real_devices())
 
 
-def sync():
-  """Launches all pending graph operations."""
-  xm.mark_step()
+def sync(wait: bool = False):
+  """Launches all pending graph operations.
+  
+  Args:
+    wait (bool): whether to block the current process until the execution finished.
+
+  """
+  torch_xla._XLAC._xla_step_marker(
+      torch_xla._XLAC._xla_get_default_device(),
+      [],
+      wait=wait,
```

Comment:
Isn't `wait` currently broken? Does it actually `WaitDeviceOps`?

Reply:
let me check, if not let me fix in a follow up pr.